spec/controllers/settings_controller_spec.rb

```diff
@@ -2,4 +2,59 @@
 
 describe SettingsController, type: :controller do
 
+  let(:user) { create(:user) }
+  let(:attributes) { attributes_for :user }
+
+  before do
+    allow(self.controller).to receive(:current_user).and_return(user)
+  end
+
+  describe "GET #index" do
+    it "assigns current_user to @settings" do
+      get :index
+      expect(assigns(:settings)).to eq(user)
+    end
+
+    it "is successful" do
+      get :index
+      expect(response).to be_success
+    end
+
+    it "renders the index view" do
+      get :index
+      expect(response).to render_template("index")
+    end
+  end
+
+  describe "POST #update" do
+    context "with valid attributes" do
+      it "updates attributes of @settings" do
+        post :update, :settings => {name: "New Name", email: "new@email.com"}
+        expect(user[:name]).to eq("New Name")
+        expect(user[:email]).to eq("new@email.com")
+      end
+
+      it "redirects to the settings page" do
+        subject { post :update, :settings => {name: "New Name", email: "new@email.com"} }
+        expect(subject).to redirect_to(settings_url)
```

Comment:
Looking at this one fail in Travis.  Subject is being set to the controller and not the response from the controller, that's why it's getting a 200 and not the 300 redirect.